Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/SendResult.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+
+namespace Backbone.Modules.Devices.Infrastructure.PushNotifications.DirectPush;
+
+public class SendResult
+{
+    public bool IsSuccess { get; private init; }
+    public bool IsFailure => !IsSuccess;
+    public SendError Error { get; private init; }
+
+    public enum FailureReason
+    {
+        InvalidHandle,
+        Unexpected
+    }
+
+    public class SendError
+    {
+        public SendError(DeviceId deviceId, FailureReason reason, string message)
+        {
+            DeviceId = deviceId;
+            Reason = reason;
+            Message = message;
+        }
+
+        public DeviceId DeviceId { get; private set; }
```

Comment:
The DeviceId should be part of the SendResult, and not of the SendError. Because a Success should of course have a DeviceId as well.